pysearch/harvester/spiders/harvester.py

```diff
@@ -0,0 +1,143 @@
+    'host': 'localhost',
+    'port': '5432',
+    'database': 'pysearch'
+}
+
+
+def db_connect():
+    """Perform database connection using database settings from settings.py. Returns sqlalchemy engine instance."""
+    return create_engine(URL(**DATABASE))
+
+
+def create_keyword_table(engine):
+    """Create keywords table."""
+    Base.metadata.drop_all(engine)
+    Base.metadata.create_all(engine)
+
+# ********************************END DATABASE******************************
+NUM_OF_OCCURANCES = 3
+
+
+class HarvestSpider(scrapy.Spider):
+    """Spider for harvesting words from a URL."""
+
+    name = "harvester"
+
+    def __init__(self, url=None, *args, **kwargs):
+        """Initialize a harvest spider."""
+        self.url = url
+        engine = db_connect()
+        create_keyword_table(engine)
+        self.Session = sessionmaker(bind=engine)
+
+    def start_requests(self):
+        """Starting place for request."""
+        yield scrapy.Request(url=self.url, callback=self.parse)
+
+    def parse(self, response):
+        """Get words from site."""
+        words = []
+        headers = []
+        title = []
+        words_in_title = []
+        words_in_headers = []
+        stop_words = get_stop_words('english')
+        h1_res = response.css('h1::text').extract()
+        h2_res = response.css('h2::text').extract()
+        h3_res = response.css('h3::text').extract()
+        h4_res = response.css('h4::text').extract()
+        h5_res = response.css('h5::text').extract()
+        h6_res = response.css('h6::text').extract()
+        title_res = response.css('title::text').extract()
+        p_res = response.css('p::text').extract()
+
+        for each in p_res:
+            words.extend(each.split())
+        for each in h1_res:
+            headers.extend(each.split())
+        for each in h2_res:
+            headers.extend(each.split())
+        for each in h3_res:
+            headers.extend(each.split())
+        for each in h4_res:
+            headers.extend(each.split())
+        for each in h5_res:
+            headers.extend(each.split())
+        for each in h6_res:
+            headers.extend(each.split())
+        for each in title_res:
+            title.extend(each.split())
+        words = lower_list(words)
+        headers = lower_list(headers)
+        title = lower_list(title)
+        for word in words:
+            if word in headers and word not in stop_words:
+                words_in_headers.append(word)
+            if word in title and word not in stop_words:
+                words_in_title.append(word)
+        word_count = collections.Counter(words)
+        for key, count in dropwhile(lambda key_count: key_count[1] >= NUM_OF_OCCURANCES, word_count.most_common()):
+            del word_count[key]
+        for key in list(word_count.keys()):
+            if key in stop_words or key == '' or key == ' ':
+                del word_count[key]
+
+        to_add = []
+        session = self.Session()
```

Comment:
This is a lot of logic for one method. You could've broken up this monolith of a method into several smaller, more testable functions.